core/interception.js

```diff
@@ -145,340 +172,446 @@ wsHook.after = function(messageEvent, url)
+        var binary = '';
+        var bytes = new Uint8Array(buffer);
+        var len = bytes.byteLength;
+        for (var i = 0; i < len; i++)
+        {
+            binary += String.fromCharCode(bytes[i]);
+        }
+        return btoa(binary);
+    }
 
-        for (var i = 0; i < children.length; i++) 
+
+    NodeHandler.isReceivedNodeAllowed = function (node, tag)
+    {
+        try
         {
-            var child = children[i];
+            var children = nodeReader.children(node);
+            var tag = nodeReader.tag(node);
+            var type = nodeReader.attr("type", node);
+            if (tag != "action") return true;
+            if (!Array.isArray(children)) return true;
 
-            var action = child[0];
-            var data = child[1];
 
-            var message = parseMessage(children[i]);
 
-            var messageRevokeValue = messageTypes.Message.ProtocolMessage.TYPE.REVOKE;
-            if (message && message.message && message.message.protocolMessage && message.message.protocolMessage.type == messageRevokeValue)
+            for (var i = 0; i < children.length; i++)
             {
-                // someone deleted a message, block
-                console.log("WhatsIncognito: --- Blocking message REVOKE action! ---");
-                return false;
-            }            
-        }
-    }
-    catch (exception)
-    {
-        console.error("WhatsIncognito: Allowing WA packet due to exception:");
-        console.error(exception);
-        return true;
-    }
-    
-    return true;
-}
+                var child = children[i];
 
-NodeHandler.manipulateReceivedNode = function(node)
-{
-    var messages = [];
-    var children = nodeReader.children(node);
-    var tag = nodeReader.tag(node);
-    var type = nodeReader.attr("type", node);
+                var action = child[0];
+                var data = child[1];
 
-    if (Array.isArray(children)) 
-    {
-        for (var i = 0; i < children.length; i++) 
-        {
-            var child = children[i];
-            var action = child[0];
-            var data = child[1];
+                var message = parseMessage(children[i]);
+                var messageRevokeValue = messageTypes.Message.ProtocolMessage.TYPE.REVOKE;
+                if (message && message.message && message.message.protocolMessage && message.message.protocolMessage.type == messageRevokeValue)
+                {
+                    // someone deleted a message, block
+                    if (saveDeletedMsgsHook)
+                    {
+                        const chat = getChatByJID(message.key.remoteJid)
+                        const msgs = chat.msgs.models
+                        let deletedMsgContents = {}
+                        for (let i = 0; i < msgs.length; i++)
+                        {
+                            if (msgs[i].id.id == message.message.protocolMessage.key.id)
+                            {
+
+                                const runMe = async () =>
```

Comment:
Perhaps the `runMe` function can be renamed and moved outside of `isReceivedNodeAllowed`

Reply:
moved to the bottom along with the deletedDB functions as well